js/src/components/app-text/index.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import './index.scss';
+
+/**
+ * Renders a paragraph with some predefined styles based on the variant
+ *
+ *
+ * ```jsx
+ * <AppText variant="title-small">
+ * 		My text
+ * </AppText>
+ * ```
+ *
+ * @param {Object} props Component props
+ * @param {string} [props.variant] The variant to use for the text
+ * @param {string} [props.className] Custom classname for this component
+ * @param {keyof JSX.IntrinsicElements} [props.as] as Custom HTML tag for the component (by default <p>)
```

Comment:
Looks like the current JSDoc config supports `keyof` but the `npm run doc:tracking` script doesn't. Probably it needs more investigation.

Reply:
Thanks for catching this @eason9487. 

Is this a blocker? Or it's possible to set this prop as "string"  and continue investigation in future PR on the `doc:tracking` problem?

Reply:
Alternative options are:

- Fix the JSDOC tracking to support keyof. Ideal but it will need more time and me getting familiar with that plugin. So it will delay everything a bit. 
- Add just the common tags like 'p'|'h1'|'h2'...|'h6'|'div'|'span'...
- Like the previous option, but adding all the tags.

Reply:
> Is this a blocker?

Since `{keyof JSX.IntrinsicElements}` is very precise, so this made me hesitate.

> Or it's possible to set this prop as "string" and continue investigation in future PR on the `doc:tracking` problem?

I vote for this one since it appears to be the most ideal option at present. Using `string` type and maybe adding one more sentence like ", e.g. h2, span." to its description would be very enough for devs to understand this prop quickly.